src/cbor/common.c

```diff
@@ -158,3 +158,101 @@ cbor_item_t *cbor_move(cbor_item_t *item) {
   item->refcount--;
   return item;
 }
+
+bool cbor_equal(cbor_item_t *item1, cbor_item_t *item2) {
+  size_t i;
+  cbor_item_t * cur_item1, * cur_item2;
+  bool ret;
+  
+  if (item1 != NULL && item2 != NULL) {
+    if (item1->type != item2->type) {
+      return false;
+    } else {
+      switch (item1->type) {
+        case CBOR_TYPE_UINT:
+        case CBOR_TYPE_NEGINT:
+          if (cbor_int_get_width(item1) != cbor_int_get_width(item2)) {
+            return false;
+          } else {
+            if (cbor_int_get_width(item1) == CBOR_INT_8)
+              return !memcmp(item1->data, item2->data, sizeof(uint8_t))?true:false;
+            else if (cbor_int_get_width(item1) == CBOR_INT_16)
+              return !memcmp(item1->data, item2->data, sizeof(uint16_t))?true:false;
+            else if (cbor_int_get_width(item1) == CBOR_INT_32)
+              return !memcmp(item1->data, item2->data, sizeof(uint32_t))?true:false;
+            else if (cbor_int_get_width(item1) == CBOR_INT_64)
+              return !memcmp(item1->data, item2->data, sizeof(uint64_t))?true:false;
+            else
+              return false;
+          }
+          break;
+        case CBOR_TYPE_BYTESTRING:
+          if (cbor_bytestring_length(item1) != cbor_bytestring_length(item2))
+            return false;
+          else
+            return !memcmp(item1->data, item2->data, cbor_bytestring_length(item1))?true:false;
```

Comment:
This doesn't work for indefinite (chunked) string

Reply:
Nitpick: No need for the ternary check, the value can be returned directly

Reply:
no problem, I thought that returning explicitly `true` and `false` would have been better for third-party readers and our future self but it's a bit overkill I guess.

Reply:
done